03-devops-overview.Rmd

```diff
@@ -0,0 +1,183 @@
+
+To work with internal DevOps Repos you will need an account with Basic access level.
+There is a form to sort out DevOps account levels and access on the IT service portal - search for DevOps.
+You will also need someone to give you permission to see the DevOps project that you're working on.
+For information about how to do this, pleae read on.
+If you need a new project area on DevOps, use the same IT service portal form as for account access.
+
+## Theory 
+
+At a high level when you are using DevOps (or any type of remote) you are creating a central copy of your code on a server (the DevOps site). 
+
+The aim of any project is to get your final complete code on to the main branch of the server copy. Analysts will still make and run all their changes locally but will then use the concepts of *pushing*, *pulling* and *pull requesting* to get the DevOps version up to date.
+
+Having this central repository and the approval processes that DevOps provides means that multiple analysts can work on a project at the same time and systematically merge all their changes into the final version after code review.
+
+The following diagram illustrates the concept of server and local copies. 
+
+![](www/distributed_version_control.png)
+
+## Creating a DevOps repository for a new project
+
+To set up a repository on DevOps you need to:
+
+1. Open the DevOps website.
+2. Navigate to repositories.
+3. Find the current repository name at the top of page and click on it to bring up a dropdown menu.
+4. Click "add respository" and give it a suitable name.
+
+This will create you a Git repo on DevOps. By default it will give you a main branch and a README file.
+
+### Clone to your computer
+
+Cloning a repository means that you:
+
+1. Get a copy of it on your computer.
+2. Your computer stores where to find the online copy so it can talk to the online one when you ask it to.
+
+To clone the blank repository to your computer you need to get the repo's URL.
+It is easiest to do this by:
+
+1. Going to "Files" in the "Repos" section of DevOps.
+2. Ensuring you are in the right repository by checking the name in the banner at the top of the screen.
+3. Clicking the clone button in the top right-hand corner.
+4. Copying the URL.
+
+Once you have copied the URL, you need to open git bash on your computer and clone the repository using the URL.
+When you open git bash you should do so in the folder where you want the repo to be.
+Ideally this will be on your C drive, outside of OneDrive of Teams because git repositories don't work well if OneDrive
+or Teams are syncing them.
+There is no need for the additional syncing from OneDrive or Teams because DevOps provides the online back-up that others can access.
+This means that any general rules than work should be on OneDrive or Teams to be backed up does not apply to repos because DevOps provides the back-up *as long as you push your work regularly and at least once a day*.
```

Comment:
Maybe reword this? It's a bit confusing:

'As long as you push your work regularly and at least once a day, DevOps provides the back-up and other general rules that are usually applied to files on OneDrive or Teams are not necessary.'

Reply:
I've gone for some hybrid of your suggestion and what I had before. I do really like your suggestion of bring the "as long as..." to the start rather than the end.